rust/parquet/src/arrow/array_reader.rs

```diff
@@ -503,7 +492,14 @@ where
             data_buffer.into_iter().map(Some).collect()
         };
 
-        self.converter.convert(data)
+        // TODO: I did this quickly without thinking through it, there might be edge cases to consider
+        let mut array = self.converter.convert(data)?;
+
+        if let ArrowType::Dictionary(_, _) = self.data_type {
+            array = arrow::compute::cast(&array, &self.data_type)?;
```

Comment:
This is really the part I just couldn't see... the code is so much better now!!!!!